src/components/functionBased/TodosList.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import TodoItem from './TodoItem';
+
+const TodosList = (props) => {
+  const {
+    todos, handleChangeProps, deleteTodoProps, setUpdate,
+  } = props;
+
+  return (
+    <ul
+      className="w-full max-h-80 overflow-y-auto lg:w-11/12 flex flex-col align-center self-center mx-auto"
+    >
+      {todos.map((todo) => (
+        <TodoItem
+          key={todo.id}
+          todo={todo}
+          handleChangeProps={handleChangeProps}
+          deleteTodoProps={deleteTodoProps}
+          setUpdate={setUpdate}
+        />
+      ))}
+    </ul>
+  );
+};
+
+TodosList.propTypes = {
+  todos: PropTypes.arrayOf(PropTypes.arrayOf).isRequired,
+  handleChangeProps: PropTypes.func.isRequired,
+  deleteTodoProps: PropTypes.func.isRequired,
+  setUpdate: PropTypes.func.isRequired,
+};
+
```

Comment:
The errors I am getting are related the highlighted code above:

```
react-jsx-dev-runtime.development.js:117 Warning: Failed prop type: The prop `todos` is marked as required in `TodosList`, but its value is `null`.
```

```
Uncaught TypeError: Cannot read properties of null (reading 'map')
```

Reply:
@mirouhml, Did you pull the new changes?

Reply:
I see this one, it is because your local storage is empty.
I will look at a way to fix this.
Now I get your point, thanks :pray:

Reply:
Sorry for not responding earlier, yea I did pull the new changes, let me know when you fix it.

Reply:
I have added a conditional display, so the component only maps through the array if it is not empty.
If this still doesn't work, I will try dropping the required property.
On my side I am getting no errors but if you find any, please let me know.

Reply:
@jssol thanks for the update, I tested again, the issue is still there, I think it's because the local storage is empty so why don't you test first whether the storage is empty or not in the TodoContainer component before proceeding with the other steps? I think that would be a better solution.

Reply:
@mirouhml This bug is a silly one, I will try my project on someone else's computer because I am not getting the same error as you.

Reply:
@jssol try emptying your local storage using inspect in the browser, that will allow you to see the error.

Reply:
@mirouhml, I have tried on another computer and I got different errors :cockroach:
We then removed the package-lock.json and the node_modules dir and reinstalled it again.
This solved the errors we were getting on the other computer :heavy_check_mark:
Can you try doing the same if you have time?
I will learn more about this as these errors may be occurring because of different node versions.
My project runs on `v16.13.1`
Thanks again for your time :pray:

Reply:
Hello @jssol,
I did try reinstalling everything already the issue still occurred :confused:, also please check my previous message, that should explain why the issue is happening :monocle_face:, if you have more questions regarding that let me know.